common-content/en/module/complexity/big-o/index.md

```diff
@@ -163,8 +174,12 @@ You will explore this theory in your backlog, but you will find that you already
   [LABEL=Quadratic Time]
 - Everyone at a party shaking hands with everyone else. If you double the number of people (n), the number of handshakes increases much faster (roughly n \* n). This is like nesting a loop inside a loop.
   [LABEL=Exponential Time]
-- Trying every possible combination to unlock a password. Each extra character dramatically increases the possibilities. This is like naive recursion;  we'll talk about this more later.
+- Trying every possible combination to unlock a password. Each extra character dramatically increases the possibilities.
   {{< /label-items >}}
 
 > [!TIP]
 >  Big-O notation also describes space complexity (how memory use grows). Sometimes an algorithm's time complexity is different from its space complexity. We have focused on time here, but you'll meet space complexity analysis in the assigned reading. 
+
+Big-O notation is focused on the _trend_ of growth, not the exact growth.
+
+If we think about strings, one character may take up one byte or four. If we think about doubling the length of the string, we don't worry about what characters are actually in the string. We think about the _trend_ - the string will take _about_ twice as much space. If the string only contained four-byte characters, and we only added one-byte characters, the string is still growing linearly, even though it may not take exactly double the space.
```

Comment:
I think this might belong in the coursework ?

The trend is explained in the linked texts btw!

Reply:
I think the reading is quite vague on this - I think the only coverage is "Big O notation is used to express the worst-case scenario of an algorithm's runtime performance, abstracting away constants and lower-order terms to focus only on the most significant factors that affect the running time." which is easy to miss or mis-understand, so figured it was worth re-stating. Rephrased repetition is helpful...

Reply:
Ok, let's just simplify this a bit then? I made a tiny edit